compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -609,6 +610,7 @@ fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
             &dead_store_elimination::DeadStoreElimination::Initial,
             &gvn::GVN,
             &simplify::SimplifyLocals::AfterGVN,
+            &merge_branches::MergeBranchSimplification,
```

Comment:
Why here in the pipeline?

Reply:
This needs to run after GVN. I expect `SimplifyLocals::AfterGVN` can simplify the analysis in `MergeBranchSimplification`. Additionally, after `MergeBranchSimplification` simplifies the CFG, it will benefit subsequent passes that are related to the CFG.